src/ecalc/libraries/libecalc/common/libecalc/core/graph_result.py

```diff
@@ -204,31 +204,50 @@ def _parse_emissions(emissions: Dict[str, EmissionResult]) -> Dict[str, libecalc
         }
 
     @staticmethod
-    def extract_pressures_from_models(
+    def extract_input_compressor_pressures_from_models(
         energy_usage_model: Dict[datetime, Any],
         pressure_type: CompressorPressureType,
+        name: str,
     ) -> TemporalModel[Expression]:
         """Get temporal model for compressor inlet- and outlet pressures.
         The pressures are the actual pressures defined by user in input.
 
         :param energy_usage_model: Temporal energy model
         :param pressure_type: Compressor pressure type, inlet- or outlet
+        :param name: name of compressor
         :return: Temporal model with pressures as expressions
         """
         evaluated_temporal_energy_usage_models = {}
         for period, model in TemporalModel(energy_usage_model).items():
-            pressures = model.suction_pressure
+            if isinstance(model, CompressorSystemConsumerFunction):
+                for compressor, operational_setting in zip(model.compressors, model.operational_settings):
```

Comment:
does this handle if the operational settings is/has temporal models as well? Is that possible?

Reply:
That was the intention at least. Thought I have checked that it actually does that - but need to double check:-)

Reply:
And more detailed:
![image](https://github.com/equinor/ecalc/assets/70878501/dcb9e172-4b4d-4d49-abae-e63fb2c18cb2)

Reply:
So as far as I can see, the function handles temporal models both for compressor systems, and for compressors

Reply:
nice. My mistake, thought we could have temporal models on operational settings only. That is not the case. I trust you :) ;)